optuna_dashboard/_objective_form_widget.py

```diff
@@ -82,51 +91,78 @@ def to_dict(self) -> SliderWidgetJSON:
             "max": self.max,
             "step": self.step,
             "labels": labels,
+            "user_attr_key": self.user_attr_key,
         }
 
 
 @dataclass
-class ObjectiveTextInputWidget:
+class TextInputWidget:
     description: Optional[str] = None
+    user_attr_key: Optional[str] = None
 
     def to_dict(self) -> TextInputWidgetJSON:
         return {
             "type": "text",
             "description": self.description,
+            "user_attr_key": self.user_attr_key,
         }
 
 
 @dataclass
 class ObjectiveUserAttrRef:
     key: str
+    user_attr_key: Optional[str] = None
 
     def to_dict(self) -> UserAttrRefJSON:
         return {
             "type": "user_attr",
             "key": self.key,
+            "user_attr_key": self.user_attr_key,
         }
 
 
-ObjectiveFormWidget = Union[
-    ObjectiveChoiceWidget, ObjectiveSliderWidget, ObjectiveTextInputWidget, ObjectiveUserAttrRef
-]
-SYSTEM_ATTR_KEY = "dashboard:objective_form_widgets:v1"
+ObjectiveFormWidget = Union[ChoiceWidget, SliderWidget, TextInputWidget, ObjectiveUserAttrRef]
+# For backward compatibility.
+ObjectiveChoiceWidget = ChoiceWidget
+ObjectiveSliderWidget = SliderWidget
+ObjectiveTextInputWidget = TextInputWidget
+FORM_WIDGETS_KEY = "dashboard:form_widgets:v2"
+FORM_WIDGETS_OUTPUT_TYPE_KEY = "dashboard:form_widgets_output_type:v2"
```

Comment:
I am wondering if `FORM_WIDGETS_OUTPUT_TYPE_KEY` could be included in `FORM_WIDGETS_KEY` as follows.
https://github.com/optuna/optuna-dashboard/commit/d0cb33eb2ab43404b2b9e749ecfb459982c2c1bc

Reply:
Thanks! I refactored form_widgets_output_type according to your comment.